catalyst-gateway/bin/src/service/common/types/cardano/asset_value.rs

```diff
@@ -104,6 +104,12 @@ impl TryFrom<num_bigint::BigInt> for AssetValue {
     }
 }
 
+impl From<i128> for AssetValue {
+    fn from(value: i128) -> Self {
```

Comment:
Needs to be a `try_from`,
It needs to check the value converted is valid.  Do not assume that it is.  Not all i128 values are valid Asset Values.
Use the `is_valid` functions to check.
If this conversion fails, in the caller, log an error and refuse to add the row.  But do not fail to produce a response.
The reason is, that this should NOT happen if the ledger validates properly, but we can't rely on the ledger being 100% correct.
For our code to be robust we need to check it.
And we need to not fail if we get this problem.